LungCancerDetection/README.md

```diff
@@ -0,0 +1,109 @@
+
+Once the fast trees model is trained, it can be operationalized to classify cancerous scans for other patients using a web app.
+
+In the next sections we will explain how to execute this system inside SQL. All the data, models and resulting features are stored and queried in different tables of a SQL database. There are 3 main processes: featurization, training and scoring.
+
+
+## Installation
+
+The installation process can be found [here](INSTALL.md).
+
+## Preprocessing
+
+We have to download the data from [kaggle dataset](https://www.kaggle.com/c/data-science-bowl-2017/data). The images are in [DICOM format](https://en.wikipedia.org/wiki/DICOM) and consist of a group of slices of the thorax of each patient as it is shown in the following figure:
+
+<p align="center">
+	<img src="https://msdnshared.blob.core.windows.net/media/2017/02/021717_1842_QuickStartG2.png" alt="lung cancer scans" width="30%"/>
+	<img src="https://msdnshared.blob.core.windows.net/media/2017/02/021717_1842_QuickStartG3.png" alt="lung cancer scans" width="30%"/>
+</p>
+
+We are going to upload the images to SQL. The reason for doing this, instead of reading the images directly from disk, is because we want to simulate an scenario where all the data is already in SQL. For demo purposes we are going to use a small subset of the images, they can be found in [stage1_labels_partial.csv](data/stage1_labels_partial.csv). This subset consists of 200 patients out of 1595. The complete patient info is [stage1_labels.csv](data/stage1_labels.csv).
+
+The first step is to create in SQL Server a database called `LungCancer`. We also require that FileTables are enabled on the database. This is for working with image data in SQL Server. Follow the instructions [here](https://docs.microsoft.com/en-us/sql/relational-databases/blob/enable-the-prerequisites-for-filetable) to set it up. Execute [file_table_creation.sql](sql/file_table_creation.sql) in order to create this database and allow it to support FileTables. This script will also create a FileTable called `MriData`.
+
+In addition, SQLRUsergroup user group should be granted dbreader access on your database. Otherwise, the sp_execute_external_script won't be able to access the image files.
+
+The next step is to upload the images. First you need to put the correct paths in the file [config_preprocessing.py](preprocessing/config_preprocessing.py.template). In case you want to upload the full dataset, just uncomment `STAGE1_LABELS = os.path.join(DATA_PATH, 'stage1_labels.csv')`. In addition, you need to set `IMAGES_FOLDER` to the location of your file table. To import the images to the SQL database using the FileTable you have set up, you have to execute the script [insert_scan_images_in_sql_database.py](preprocessing/insert_scan_images_in_sql_database.py). This will take a while.
```

Comment:
not clear how to set IMAGES_FOLDER and STAGE1_FOLDER?

Reply:
IMAGES_FOLDER should be set in connection_strings.py, something like: IMAGES_FOLDER =  '\\\\Tyler-Laptop\\tylersqlserver\\DemoData\\MriData'

Reply:
STAGE1_FOLDER should be set in config_preprocessing.py and point to where the downloaded dataset has been extracted to.

Reply:
Updated